python/ray/serve/_private/router.py

```diff
@@ -260,11 +308,20 @@ async def choose_two_replicas_with_backoff(
                     extra={"log_to_stderr": False},
                 )
 
+            if len(replica_ids_with_multiplexed_model_id) > 0:
+                replica_ids_to_consider = replica_ids_with_multiplexed_model_id
+            else:
+                replica_ids_to_consider = self._replica_id_set
+
             chosen_ids = random.sample(
-                self._replica_id_set, k=min(2, len(self._replica_id_set))
+                replica_ids_to_consider, k=min(2, len(replica_ids_to_consider))
             )
             yield [self._replicas[chosen_id] for chosen_id in chosen_ids]
 
+            if len(replica_ids_with_multiplexed_model_id) > 0:
+                for chosen_id in chosen_ids:
+                    replica_ids_with_multiplexed_model_id.remove(chosen_id)
+
             await asyncio.sleep(self.backoff_sequence_s[backoff_index])
```

Comment:
During the backoff, the requested model_id information might come in, and reflected in self._multiplexed_model_id_to_replica_ids.  Is it possible we update the `replica_ids_with_multiplexed_model_id` every loop?

Reply:
Fixed this and added a test case for it: `test_no_replicas_available_then_choose_one_with_id`